app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -42,51 +44,35 @@ import com.google.android.gms.maps.GoogleMap.OnCameraIdleListener
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.*
-import com.google.android.libraries.places.api.model.Place
-import com.google.android.libraries.places.api.Places
-import com.google.android.libraries.places.widget.Autocomplete
-import com.google.android.libraries.places.widget.AutocompleteActivity
-import com.google.android.libraries.places.widget.model.AutocompleteActivityMode
 import com.google.android.material.bottomsheet.BottomSheetBehavior
-import com.google.maps.android.PolyUtil
 import com.mancj.materialsearchbar.MaterialSearchBar
+import kotlinx.android.synthetic.main.bottom_sheet_layout.*
+import kotlinx.android.synthetic.main.map_fragment.*
+import kotlinx.android.synthetic.main.search_bar_layout.*
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 import java.io.IOException
-import java.util.Locale
-import kotlin.collections.ArrayList
-import kotlin.collections.List
-import kotlin.collections.listOf
-import kotlin.collections.MutableList
-import kotlinx.android.synthetic.main.bottom_sheet_layout.bottom_sheet
-import kotlinx.android.synthetic.main.bottom_sheet_layout.radioTransportGroup
-import kotlinx.android.synthetic.main.map_fragment.buttonInstructions
-import kotlinx.android.synthetic.main.map_fragment.searchBar
-import kotlinx.android.synthetic.main.map_fragment.toggleButton
-import org.json.JSONArray
-import org.json.JSONObject
-import com.droidhats.campuscompass.helpers.Observer
-import com.droidhats.campuscompass.helpers.Subject
 
 class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListener,
-    GoogleMap.OnPolygonClickListener, CalendarFragment.OnCalendarEventClickListener,
-    OnCameraIdleListener, Subject{
+    GoogleMap.OnPolygonClickListener, CalendarFragment.OnCalendarEventClickListener, SearchAdapter.OnSearchResultClickListener, OnCameraIdleListener,
+    Subject {
 
     private lateinit var map: GoogleMap
     private lateinit var fusedLocationClient: FusedLocationProviderClient
     private lateinit var lastLocation: Location
     private lateinit var locationCallback: LocationCallback
     private lateinit var locationRequest: LocationRequest
-    private val observerList = mutableListOf<Observer>()
+    private val observerList = mutableListOf<com.droidhats.campuscompass.helpers.Observer?>()
```

Comment:
We can possibly add an import statement like `import com.droidhats.campuscompass.helpers.Observer as ModifiedObserver` and then replace `com.droidhats.campuscompass.helpers.Observer` with `ModifiedObserver` through out the code